impeller/aiks/canvas.cc

```diff
@@ -650,14 +650,30 @@ void Canvas::DrawImageRect(const std::shared_ptr<Image>& image,
     return;
   }
 
-  auto contents = TextureContents::MakeRect(dest);
-  contents->SetTexture(image->GetTexture());
-  contents->SetSourceRect(source);
-  contents->SetStrictSourceRect(src_rect_constraint ==
-                                SourceRectConstraint::kStrict);
-  contents->SetSamplerDescriptor(std::move(sampler));
-  contents->SetOpacity(paint.color.alpha);
-  contents->SetDeferApplyingOpacity(paint.HasColorFilter());
+  auto texture_contents = TextureContents::MakeRect(dest);
+  texture_contents->SetTexture(image->GetTexture());
+  texture_contents->SetSourceRect(source);
+  texture_contents->SetStrictSourceRect(src_rect_constraint ==
+                                        SourceRectConstraint::kStrict);
+  texture_contents->SetSamplerDescriptor(std::move(sampler));
+  texture_contents->SetOpacity(paint.color.alpha);
+  texture_contents->SetDeferApplyingOpacity(paint.HasColorFilter());
+
+  bool needs_color_filter = paint.HasColorFilter();
+  if (needs_color_filter) {
+    auto color_filter = paint.GetColorFilter();
+    if (texture_contents->ApplyColorFilter(
```

Comment:
I don't think this will ever pass? This is for automagically filtering solid colors or gradients, things that exist on host memory.

Reply:
Done, removed it.